File: clients/src/bin/simple_transfer_instruction.rs
use anyhow::Result;
use solana_sdk::{
    native_token::LAMPORTS_PER_SOL, signature::Signer, signer::keypair::Keypair, system_instruction,
};

#[tokio::main]
async fn main() -> Result<()> {
    // Generate sender and recipient keypairs
    let sender = Keypair::new();
    let recipient = Keypair::new();

    // Define the amount to transfer
    let transfer_amount = LAMPORTS_PER_SOL / 100; // 0.01 SOL

    // Create a transfer instruction for transferring SOL from sender to recipient
    let transfer_instruction =
        system_instruction::transfer(&sender.pubkey(), &recipient.pubkey(), transfer_amount);

    println!("{:#?}", transfer_instruction);

    Ok(())
}

File: scripts/simple_transfer_instruction.ts
import {
    LAMPORTS_PER_SOL,
    SystemProgram,
    Transaction,
    Keypair
  } from "@solana/web3.js";
  
  // Generate sender and recipient keypairs
  const sender = Keypair.generate(); // Keypair.generate() is equivalent to new Keypair()
  const recipient = new Keypair();
  
  // Define the amount to transfer
  const transferAmount = 0.01; // 0.01 SOL
  
  // Create a transfer instruction for transferring SOL from sender to recipient
  // by calling the system program
  const transferInstruction = SystemProgram.transfer({
    fromPubkey: sender.publicKey,
    toPubkey: recipient.publicKey,
    lamports: transferAmount * LAMPORTS_PER_SOL // Convert transferAmount to lamports
  });
  
  console.log(JSON.stringify(transferInstruction, null, 2));

  